### ISSUE TITLE
is it possible the docker-compose.yml is missing a ports: - "3306:3306"

### ISSUE THREAD

**Issue by nothingtoseeherehub** [CTFd/CTFd#266]
db:
    image: mysql
    ports:
      - "3306:3306"
    environment:
      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - .data/mysql:/var/lib/mysql

Without the ports entry in the db section, I'm not able to bring up a functioning page.  The webpage shows, but nothing functions that would require db connectivity.  The stdout shows errors with sql connection without the ports entry.

**Comment by ColdHeat** [comment]
You shouldn't need to specify ports. I think this might be because the mysql version isn't pinned?

**Comment by ColdHeat** [comment]
Hmm I suspect this is because of some shenanigans that Docker or MySQL introduced where you can't login as root remotely or something. 

I'll debug it after I finish this other CTFd feature. In the meantime you can just use `serve.py`.

**Comment by ColdHeat** [comment]
Hey @nothingtoseeherehub can you try pulling from master? 

If that doesn't fix it for you, please re-open :)

**Comment by nothingtoseeherehub** [comment]
New issue with the update--
<snip>
 RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version
--
<endsnip>
background:
amazon t2-micro . . .
Into a new directory, and removing all pre-existing images/containers
git clone https://github.com/CTFd/CTFd
<snip>

docker-compose up
Building ctfd
Step 1/10 : FROM python:2.7-alpine
 ---> 3dd614730c9c
Step 2/10 : RUN apk update &&     apk add python python-dev libffi-dev gcc make musl-dev py-pip mysql-client
 ---> Running in eacb33963ec4
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
v3.4.6-129-g6d398ff [http://dl-cdn.alpinelinux.org/alpine/v3.4/main]
v3.4.6-83-g67e50bc [http://dl-cdn.alpinelinux.org/alpine/v3.4/community]
OK: 5974 distinct packages available
(1/24) Installing binutils-libs (2.26-r1)
(2/24) Installing binutils (2.26-r1)
(3/24) Installing gmp (6.1.0-r0)
(4/24) Installing isl (0.14.1-r0)
(5/24) Installing libgomp (5.3.0-r0)
(6/24) Installing libatomic (5.3.0-r0)
(7/24) Installing libgcc (5.3.0-r0)
(8/24) Installing pkgconf (0.9.12-r0)
(9/24) Installing pkgconfig (0.25-r1)
(10/24) Installing mpfr3 (3.1.2-r0)
(11/24) Installing mpc1 (1.0.3-r0)
(12/24) Installing libstdc++ (5.3.0-r0)
(13/24) Installing gcc (5.3.0-r0)
(14/24) Installing libffi (3.2.1-r2)
(15/24) Installing libffi-dev (3.2.1-r2)
(16/24) Installing make (4.1-r1)
(17/24) Installing musl-dev (1.1.14-r14)
(18/24) Installing mariadb-common (10.1.22-r0)
(19/24) Installing mariadb-client (10.1.22-r0)
(20/24) Installing mysql-client (10.1.22-r0)
(21/24) Installing python (2.7.12-r0)
(22/24) Installing py-setuptools (20.8.0-r0)
(23/24) Installing py-pip (8.1.2-r0)
(24/24) Installing python-dev (2.7.12-r0)
Executing busybox-1.24.2-r13.trigger
OK: 196 MiB in 56 packages
 ---> 91f43105917b
Removing intermediate container eacb33963ec4
Step 3/10 : RUN mkdir -p /opt/CTFd
 ---> Running in 11d834ed77fd
 ---> bf1118736b06
Removing intermediate container 11d834ed77fd
Step 4/10 : COPY . /opt/CTFd
 ---> 79d47c3c079c
Removing intermediate container 8f7a4431a3ed
Step 5/10 : WORKDIR /opt/CTFd
 ---> efcab14bb576
Removing intermediate container f454a4b0ac77
Step 6/10 : VOLUME /opt/CTFd
 ---> Running in 8c2c90a377a4
 ---> 86f852db2ecb
Removing intermediate container 8c2c90a377a4
Step 7/10 : RUN pip install -r requirements.txt
 ---> Running in 55f9062e6669
Collecting Flask==0.12 (from -r requirements.txt (line 1))
  Downloading Flask-0.12-py2.py3-none-any.whl (82kB)
Collecting Flask-SQLAlchemy==2.2 (from -r requirements.txt (line 2))
  Downloading Flask_SQLAlchemy-2.2-py2.py3-none-any.whl
Collecting Flask-Session==0.3.1 (from -r requirements.txt (line 3))
  Downloading Flask_Session-0.3.1-py2.py3-none-any.whl
Collecting Flask-Caching==1.2.0 (from -r requirements.txt (line 4))
  Downloading Flask_Caching-1.2.0-py2.py3-none-any.whl
Collecting Flask-Migrate==2.0.3 (from -r requirements.txt (line 5))
  Downloading Flask-Migrate-2.0.3.tar.gz
Collecting SQLAlchemy==1.1.6 (from -r requirements.txt (line 6))
  Downloading SQLAlchemy-1.1.6.tar.gz (5.2MB)
Collecting SQLAlchemy-Utils==0.32.12 (from -r requirements.txt (line 7))
  Downloading SQLAlchemy-Utils-0.32.12.tar.gz (119kB)
Collecting passlib==1.7.1 (from -r requirements.txt (line 8))
  Downloading passlib-1.7.1-py2.py3-none-any.whl (498kB)
Collecting bcrypt==3.1.3 (from -r requirements.txt (line 9))
  Downloading bcrypt-3.1.3.tar.gz (40kB)
Requirement already satisfied: six==1.10.0 in /usr/local/lib/python2.7/site-packages (from -r requirements.txt (line 10))
Collecting itsdangerous==0.24 (from -r requirements.txt (line 11))
  Downloading itsdangerous-0.24.tar.gz (46kB)
Collecting requests==2.13.0 (from -r requirements.txt (line 12))
  Downloading requests-2.13.0-py2.py3-none-any.whl (584kB)
Collecting PyMySQL==0.7.10 (from -r requirements.txt (line 13))
  Downloading PyMySQL-0.7.10-py2.py3-none-any.whl (78kB)
Collecting gunicorn==19.7.0 (from -r requirements.txt (line 14))
  Downloading gunicorn-19.7.0-py2.py3-none-any.whl (112kB)
Collecting dataset==0.8.0 (from -r requirements.txt (line 15))
  Downloading dataset-0.8.0-py2.py3-none-any.whl
Collecting mistune==0.7.4 (from -r requirements.txt (line 16))
  Downloading mistune-0.7.4-py2.py3-none-any.whl
Collecting click>=2.0 (from Flask==0.12->-r requirements.txt (line 1))
  Downloading click-6.7-py2.py3-none-any.whl (71kB)
Collecting Jinja2>=2.4 (from Flask==0.12->-r requirements.txt (line 1))
  Downloading Jinja2-2.9.6-py2.py3-none-any.whl (340kB)
Collecting Werkzeug>=0.7 (from Flask==0.12->-r requirements.txt (line 1))
  Downloading Werkzeug-0.12.2-py2.py3-none-any.whl (312kB)
Collecting alembic>=0.6 (from Flask-Migrate==2.0.3->-r requirements.txt (line 5))
  Downloading alembic-0.9.2.tar.gz (986kB)
Collecting Flask-Script>=0.6 (from Flask-Migrate==2.0.3->-r requirements.txt (line 5))
  Downloading Flask-Script-2.0.5.tar.gz (42kB)
Collecting cffi>=1.1 (from bcrypt==3.1.3->-r requirements.txt (line 9))
  Downloading cffi-1.10.0.tar.gz (418kB)
Collecting normality>=0.3.9 (from dataset==0.8.0->-r requirements.txt (line 15))
  Downloading normality-0.4.2-py2-none-any.whl
Collecting PyYAML>=3.10 (from dataset==0.8.0->-r requirements.txt (line 15))
  Downloading PyYAML-3.12.tar.gz (253kB)
Collecting MarkupSafe>=0.23 (from Jinja2>=2.4->Flask==0.12->-r requirements.txt (line 1))
  Downloading MarkupSafe-1.0.tar.gz
Collecting Mako (from alembic>=0.6->Flask-Migrate==2.0.3->-r requirements.txt (line 5))
  Downloading Mako-1.0.6.tar.gz (575kB)
Collecting python-editor>=0.3 (from alembic>=0.6->Flask-Migrate==2.0.3->-r requirements.txt (line 5))
  Downloading python-editor-1.0.3.tar.gz
Collecting python-dateutil (from alembic>=0.6->Flask-Migrate==2.0.3->-r requirements.txt (line 5))
  Downloading python_dateutil-2.6.0-py2.py3-none-any.whl (194kB)
Collecting pycparser (from cffi>=1.1->bcrypt==3.1.3->-r requirements.txt (line 9))
  Downloading pycparser-2.17.tar.gz (231kB)
Collecting chardet (from normality>=0.3.9->dataset==0.8.0->-r requirements.txt (line 15))
  Downloading chardet-3.0.3-py2.py3-none-any.whl (133kB)
Building wheels for collected packages: Flask-Migrate, SQLAlchemy, SQLAlchemy-Utils, bcrypt, itsdangerous, alembic, Flask-Script, cffi, PyYAML, MarkupSafe, Mako, python-editor, pycparser
  Running setup.py bdist_wheel for Flask-Migrate: started
  Running setup.py bdist_wheel for Flask-Migrate: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/4f/1a/cd/241202c77554d1500b47f169a59432c33834f941e90769bf0e
  Running setup.py bdist_wheel for SQLAlchemy: started
  Running setup.py bdist_wheel for SQLAlchemy: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/c5/1d/84/168a2c298f745255527b263944d2bd7115202073ce0aed3084
  Running setup.py bdist_wheel for SQLAlchemy-Utils: started
  Running setup.py bdist_wheel for SQLAlchemy-Utils: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/14/70/65/f96264b7fd409a2cc327f16666b8f8fcef2766f32c6339c6b8
  Running setup.py bdist_wheel for bcrypt: started
  Running setup.py bdist_wheel for bcrypt: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/1f/ad/7e/83015894fb7f3b92b11b4c4d4390c8175826bf03a2765e9dc4
  Running setup.py bdist_wheel for itsdangerous: started
  Running setup.py bdist_wheel for itsdangerous: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/fc/a8/66/24d655233c757e178d45dea2de22a04c6d92766abfb741129a
  Running setup.py bdist_wheel for alembic: started
  Running setup.py bdist_wheel for alembic: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/b9/d5/8e/f0de6622033684ba94d029eabee5abda70b012db80bad13bd0
  Running setup.py bdist_wheel for Flask-Script: started
  Running setup.py bdist_wheel for Flask-Script: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/e2/ea/d8/8d114e46cef819f7d9879504a7f9cb2a88a479af2858223d9f
  Running setup.py bdist_wheel for cffi: started
  Running setup.py bdist_wheel for cffi: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/7c/c4/64/889d3d2de89188ed0fcb0b313106338f06127874911eeb4c96
  Running setup.py bdist_wheel for PyYAML: started
  Running setup.py bdist_wheel for PyYAML: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/2c/f7/79/13f3a12cd723892437c0cfbde1230ab4d82947ff7b3839a4fc
  Running setup.py bdist_wheel for MarkupSafe: started
  Running setup.py bdist_wheel for MarkupSafe: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/88/a7/30/e39a54a87bcbe25308fa3ca64e8ddc75d9b3e5afa21ee32d57
  Running setup.py bdist_wheel for Mako: started
  Running setup.py bdist_wheel for Mako: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/34/be/28/7999aa61d80a775c9e85a5a7d345777f73e87e6e30573e9551
  Running setup.py bdist_wheel for python-editor: started
  Running setup.py bdist_wheel for python-editor: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/84/d6/b8/082dc3b5cd7763f17f5500a193b6b248102217cbaa3f0a24ca
  Running setup.py bdist_wheel for pycparser: started
  Running setup.py bdist_wheel for pycparser: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/a8/0b/41/dc95621f9d3a0da7bc191b8a71f0e8182ffd3cc5f33ac55005
Successfully built Flask-Migrate SQLAlchemy SQLAlchemy-Utils bcrypt itsdangerous alembic Flask-Script cffi PyYAML MarkupSafe Mako python-editor pycparser
Installing collected packages: itsdangerous, click, MarkupSafe, Jinja2, Werkzeug, Flask, SQLAlchemy, Flask-SQLAlchemy, Flask-Session, Flask-Caching, Mako, python-editor, python-dateutil, alembic, Flask-Script, Flask-Migrate, SQLAlchemy-Utils, passlib, pycparser, cffi, bcrypt, requests, PyMySQL, gunicorn, chardet, normality, PyYAML, dataset, mistune
Successfully installed Flask-0.12 Flask-Caching-1.2.0 Flask-Migrate-2.0.3 Flask-SQLAlchemy-2.2 Flask-Script-2.0.5 Flask-Session-0.3.1 Jinja2-2.9.6 Mako-1.0.6 MarkupSafe-1.0 PyMySQL-0.7.10 PyYAML-3.12 SQLAlchemy-1.1.6 SQLAlchemy-Utils-0.32.12 Werkzeug-0.12.2 alembic-0.9.2 bcrypt-3.1.3 cffi-1.10.0 chardet-3.0.3 click-6.7 dataset-0.8.0 gunicorn-19.7.0 itsdangerous-0.24 mistune-0.7.4 normality-0.4.2 passlib-1.7.1 pycparser-2.17 python-dateutil-2.6.0 python-editor-1.0.3 requests-2.13.0
 ---> 0c17c268901f
Removing intermediate container 55f9062e6669
Step 8/10 : RUN chmod +x /opt/CTFd/docker-entrypoint.sh
 ---> Running in c2200c41a50b
 ---> e3dfd21a28ba
Removing intermediate container c2200c41a50b
Step 9/10 : EXPOSE 8000
 ---> Running in 979837ef8527
 ---> 72c8294969a1
Removing intermediate container 979837ef8527
Step 10/10 : ENTRYPOINT /opt/CTFd/docker-entrypoint.sh
 ---> Running in be44c9e77d81
 ---> 59a696b466c1
Removing intermediate container be44c9e77d81
Successfully built 59a696b466c1
WARNING: Image for service ctfd was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating ctfd_db_1 ... 
Creating ctfd_db_1 ... done
Creating ctfd_ctfd_1 ... 
Creating ctfd_ctfd_1 ... done
Attaching to ctfd_db_1, ctfd_ctfd_1
ctfd_1  | Waiting on MySQL
db_1    | Initializing database
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] /usr/sbin/mysqld (mysqld 10.1.24-MariaDB-1~jessie) starting as process 58 ...
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Using mutexes to ref count buffer pool pages
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: The InnoDB memory heap is disabled
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Compressed tables use zlib 1.2.8
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Using Linux native AIO
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Using SSE crc32 instructions
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Initializing buffer pool, size = 256.0M
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Completed initialization of buffer pool
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: The first specified data file ./ibdata1 did not exist: a new database to be created!
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Setting file ./ibdata1 size to 12 MB
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Setting log file ./ib_logfile101 size to 48 MB
db_1    | 2017-06-03 21:04:14 140532883478464 [Note] InnoDB: Setting log file ./ib_logfile1 size to 48 MB
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
db_1    | 2017-06-03 21:04:15 140532883478464 [Warning] InnoDB: New log files created, LSN=45883
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Doublewrite buffer not found: creating new
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Doublewrite buffer created
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: 128 rollback segment(s) are active.
db_1    | 2017-06-03 21:04:15 140532883478464 [Warning] InnoDB: Creating foreign key constraint system tables.
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Foreign key constraint system tables created
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Creating tablespace and datafile system tables.
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Tablespace and datafile system tables created.
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB: Waiting for purge to start
db_1    | 2017-06-03 21:04:15 140532883478464 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.36-82.0 started; log sequence number 0
db_1    | 2017-06-03 21:04:15 140532131006208 [Note] InnoDB: Dumping buffer pool(s) not yet started
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] /usr/sbin/mysqld (mysqld 10.1.24-MariaDB-1~jessie) starting as process 87 ...
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Using mutexes to ref count buffer pool pages
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: The InnoDB memory heap is disabled
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Compressed tables use zlib 1.2.8
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Using Linux native AIO
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Using SSE crc32 instructions
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Initializing buffer pool, size = 256.0M
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Completed initialization of buffer pool
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Highest supported file format is Barracuda.
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: 128 rollback segment(s) are active.
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB: Waiting for purge to start
db_1    | 2017-06-03 21:04:18 140197043705792 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.36-82.0 started; log sequence number 1616799
db_1    | 2017-06-03 21:04:18 140196288890624 [Note] InnoDB: Dumping buffer pool(s) not yet started
db_1    | 2017-06-03 21:04:21 140522240001984 [Note] /usr/sbin/mysqld (mysqld 10.1.24-MariaDB-1~jessie) starting as process 116 ...
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Using mutexes to ref count buffer pool pages
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: The InnoDB memory heap is disabled
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Compressed tables use zlib 1.2.8
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Using Linux native AIO
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Using SSE crc32 instructions
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Initializing buffer pool, size = 256.0M
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Completed initialization of buffer pool
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Highest supported file format is Barracuda.
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: 128 rollback segment(s) are active.
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB: Waiting for purge to start
db_1    | 2017-06-03 21:04:22 140522240001984 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.36-82.0 started; log sequence number 1616809
db_1    | 2017-06-03 21:04:22 140521485862656 [Note] InnoDB: Dumping buffer pool(s) not yet started
db_1    | 
db_1    | PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
db_1    | To do so, start the server, then issue the following commands:
db_1    | 
db_1    | '/usr/bin/mysqladmin' -u root password 'new-password'
db_1    | '/usr/bin/mysqladmin' -u root -h  password 'new-password'
db_1    | 
db_1    | Alternatively you can run:
db_1    | '/usr/bin/mysql_secure_installation'
db_1    | 
db_1    | which will also give you the option of removing the test
db_1    | databases and anonymous user created by default.  This is
db_1    | strongly recommended for production servers.
db_1    | 
db_1    | See the MariaDB Knowledgebase at http://mariadb.com/kb or the
db_1    | MySQL manual for more instructions.
db_1    | 
db_1    | Please report any problems at http://mariadb.org/jira
db_1    | 
db_1    | The latest information about MariaDB is available at http://mariadb.org/.
db_1    | You can find additional information about the MySQL part at:
db_1    | http://dev.mysql.com
db_1    | Consider joining MariaDB's strong and vibrant community:
db_1    | https://mariadb.org/get-involved/
db_1    | 
db_1    | Database initialized
db_1    | MySQL init process in progress...
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] mysqld (mysqld 10.1.24-MariaDB-1~jessie) starting as process 149 ...
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Using mutexes to ref count buffer pool pages
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: The InnoDB memory heap is disabled
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Compressed tables use zlib 1.2.8
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Using Linux native AIO
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Using SSE crc32 instructions
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Initializing buffer pool, size = 256.0M
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Completed initialization of buffer pool
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Highest supported file format is Barracuda.
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: 128 rollback segment(s) are active.
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB: Waiting for purge to start
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.36-82.0 started; log sequence number 1616819
db_1    | 2017-06-03 21:04:25 139857833723648 [Note] InnoDB: Dumping buffer pool(s) not yet started
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] Plugin 'FEEDBACK' is disabled.
db_1    | 2017-06-03 21:04:25 139858585429952 [Warning] 'user' entry 'root@0a64b5bc2fc9' ignored in --skip-name-resolve mode.
db_1    | 2017-06-03 21:04:25 139858585429952 [Warning] 'proxies_priv' entry '@% root@0a64b5bc2fc9' ignored in --skip-name-resolve mode.
db_1    | 2017-06-03 21:04:25 139858585429952 [Note] mysqld: ready for connections.
db_1    | Version: '10.1.24-MariaDB-1~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
db_1    | Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.
db_1    | 2017-06-03 21:04:30 139858584648448 [Warning] 'proxies_priv' entry '@% root@0a64b5bc2fc9' ignored in --skip-name-resolve mode.
db_1    | 2017-06-03 21:04:31 139858584648448 [Warning] 'proxies_priv' entry '@% root@0a64b5bc2fc9' ignored in --skip-name-resolve mode.
db_1    | 
db_1    | 2017-06-03 21:04:31 139858584345344 [Note] mysqld: Normal shutdown
db_1    | 
db_1    | 2017-06-03 21:04:31 139858584345344 [Note] Event Scheduler: Purging the queue. 0 events
db_1    | 2017-06-03 21:04:31 139857816938240 [Note] InnoDB: FTS optimize thread exiting.
db_1    | 2017-06-03 21:04:31 139858584345344 [Note] InnoDB: Starting shutdown...
db_1    | 2017-06-03 21:04:31 139858584345344 [Note] InnoDB: Waiting for page_cleaner to finish flushing of buffer pool
db_1    | 2017-06-03 21:04:33 139858584345344 [Note] InnoDB: Shutdown completed; log sequence number 1616829
db_1    | 2017-06-03 21:04:33 139858584345344 [Note] mysqld: Shutdown complete
db_1    | 
db_1    | 
db_1    | MySQL init process done. Ready for start up.
db_1    | 
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] mysqld (mysqld 10.1.24-MariaDB-1~jessie) starting as process 1 ...
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Using mutexes to ref count buffer pool pages
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: The InnoDB memory heap is disabled
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Compressed tables use zlib 1.2.8
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Using Linux native AIO
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Using SSE crc32 instructions
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Initializing buffer pool, size = 256.0M
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Completed initialization of buffer pool
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Highest supported file format is Barracuda.
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: 128 rollback segment(s) are active.
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB: Waiting for purge to start
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.36-82.0 started; log sequence number 1616829
db_1    | 2017-06-03 21:04:33 140129008060160 [Note] InnoDB: Dumping buffer pool(s) not yet started
db_1    | 2017-06-03 21:04:33 140129762625472 [Note] Plugin 'FEEDBACK' is disabled.
db_1    | 2017-06-03 21:04:34 140129762625472 [Note] Server socket created on IP: '::'.
db_1    | 2017-06-03 21:04:34 140129762625472 [Warning] 'proxies_priv' entry '@% root@0a64b5bc2fc9' ignored in --skip-name-resolve mode.
db_1    | 2017-06-03 21:04:34 140129762625472 [Note] mysqld: ready for connections.
db_1    | Version: '10.1.24-MariaDB-1~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_ctfd_1 exited with code 1
ctfd_1  | Waiting on MySQL
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_ctfd_1 exited with code 1
ctfd_1  | Waiting on MySQL
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_ctfd_1 exited with code 1
ctfd_1  | Waiting on MySQL
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_ctfd_1 exited with code 1
ctfd_1  | Waiting on MySQL
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_ctfd_1 exited with code 1
ctfd_1  | Waiting on MySQL
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_ctfd_1 exited with code 1
ctfd_1  | Waiting on MySQL
ctfd_1  | .....................Ready
ctfd_1  | Starting CTFd
ctfd_1  | 
ctfd_1  | Error: class uri 'gunicorn.workers.gthread.ThreadWorker' invalid or not found: 
ctfd_1  | 
ctfd_1  | [Traceback (most recent call last):
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 142, in load_class
ctfd_1  |     mod = import_module('.'.join(components))
ctfd_1  |   File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
ctfd_1  |     __import__(name)
ctfd_1  |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/gthread.py", line 37, in <module>
ctfd_1  |     """)
ctfd_1  | RuntimeError: 
ctfd_1  |     You need to install the 'futures' package to use this worker with this
ctfd_1  |     Python version.
ctfd_1  |     
ctfd_1  | ]
ctfd_1  | 
ctfd_1  | Waiting on MySQL
ctfd_1  | Ready
ctfd_1  | Starting CTFd
ctfd_1  |

**Comment by ColdHeat** [comment]
Minor issue with gunicorn @nothingtoseeherehub. Should be fixed in master now but I'm testing on my own now.

**Comment by nothingtoseeherehub** [comment]
docker exec (containername goes here) pip install futures    resolves the error, so 'yes' there are still issues with the current state of the git repository as of 17:00ish eastern june 3rd.   It missed the 'futures feature necessary for the ctfd_ctfd_1 container

**Comment by ColdHeat** [comment]
I removed the call to `--threads` which was causing the issue, you should be good to go now.

**Comment by nothingtoseeherehub** [comment]
golden... consider issue resolved.....
